# NY House Dataset


In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("NY-House-Dataset.csv")

## Знакомство с данными

In [3]:
# Размер таблицы: (строки, столбцы)
df.shape

(4801, 17)

In [4]:
# Первые 10 строк
df.head(10)

,BROKERTITLE,TYPE,PRICE,BEDS,BATH,PROPERTYSQFT,ADDRESS,STATE,MAIN_ADDRESS,ADMINISTRATIVE_AREA_LEVEL_2,LOCALITY,SUBLOCALITY,STREET_NAME,LONG_NAME,FORMATTED_ADDRESS,LATITUDE,LONGITUDE
0,Brokered by Douglas Elliman -111 Fifth Ave,Condo for sale,315000,2,2.000000,1400.000000,2 E 55th St Unit 803,"New York, NY 10022","2 E 55th St Unit 803New York, NY 10022",New York County,New York,Manhattan,East 55th Street,Regis Residence,"Regis Residence, 2 E 55th St #803, New York, N...",40.761255,-73.974483
1,Brokered by Serhant,Condo for sale,195000000,7,10.000000,17545.000000,Central Park Tower Penthouse-217 W 57th New Yo...,"New York, NY 10019",Central Park Tower Penthouse-217 W 57th New Yo...,United States,New York,New York County,New York,West 57th Street,"217 W 57th St, New York, NY 10019, USA",40.766393,-73.980991
2,Brokered by Sowae Corp,House for sale,260000,4,2.000000,2015.000000,620 Sinclair Ave,"Staten Island, NY 10312","620 Sinclair AveStaten Island, NY 10312",United States,New York,Richmond County,Staten Island,Sinclair Avenue,"620 Sinclair Ave, Staten Island, NY 10312, USA",40.541805,-74.196109
3,Brokered by COMPASS,Condo for sale,69000,3,1.000000,445.000000,2 E 55th St Unit 908W33,"Manhattan, NY 10022","2 E 55th St Unit 908W33Manhattan, NY 10022",United States,New York,New York County,New York,East 55th Street,"2 E 55th St, New York, NY 10022, USA",40.761398,-73.974613
4,Brokered by Sotheby's International Realty - E...,Townhouse for sale,55000000,7,2.373861,14175.000000,5 E 64th St,"New York, NY 10065","5 E 64th StNew York, NY 10065",United States,New York,New York County,New York,East 64th Street,"5 E 64th St, New York, NY 10065, USA",40.767224,-73.969856
5,Brokered by Sowae Corp,House for sale,690000,5,2.000000,4004.000000,584 Park Pl,"Brooklyn, NY 11238","584 Park PlBrooklyn, NY 11238",United States,New York,Kings County,Brooklyn,Park Place,"584 Park Pl, Brooklyn, NY 11238, USA",40.674363,-73.958725
6,Brokered by Douglas Elliman - 575 Madison Ave,Condo for sale,899500,2,2.000000,2184.207862,157 W 126th St Unit 1B,"New York, NY 10027","157 W 126th St Unit 1BNew York, NY 10027",New York,New York County,New York,Manhattan,157,"157 W 126th St #1b, New York, NY 10027, USA",40.809448,-73.946777
7,Brokered by Connie Profaci Realty,House for sale,16800000,8,16.000000,33000.000000,177 Benedict Rd,"Staten Island, NY 10304","177 Benedict RdStaten Island, NY 10304",United States,New York,Richmond County,Staten Island,Benedict Road,"177 Benedict Rd, Staten Island, NY 10304, USA",40.595002,-74.106424
8,Brokered by Pantiga Group Inc.,Co-op for sale,265000,1,1.000000,750.000000,875 Morrison Ave Apt 3M,"Bronx, NY 10473","875 Morrison Ave Apt 3MBronx, NY 10473",Bronx County,The Bronx,East Bronx,Morrison Avenue,Parking lot,"Parking lot, 875 Morrison Ave #3m, Bronx, NY 1...",40.821586,-73.874089
9,Brokered by CENTURY 21 MK Realty,Co-op for sale,440000,2,1.000000,978.000000,1350 Ocean Pkwy Apt 5G,"Brooklyn, NY 11230","1350 Ocean Pkwy Apt 5GBrooklyn, NY 11230",New York,Kings County,Brooklyn,Midwood,1350,"1350 Ocean Pkwy #5g, Brooklyn, NY 11230, USA",40.615738,-73.969694


In [5]:
# Случайные 5 строк
df.sample(5)

,BROKERTITLE,TYPE,PRICE,BEDS,BATH,PROPERTYSQFT,ADDRESS,STATE,MAIN_ADDRESS,ADMINISTRATIVE_AREA_LEVEL_2,LOCALITY,SUBLOCALITY,STREET_NAME,LONG_NAME,FORMATTED_ADDRESS,LATITUDE,LONGITUDE
3830,Brokered by Amorelli Realty LLC,House for sale,899000,3,1.0,2184.207862,32-50 53 Pl,"Woodside, NY 11377","32-50 53 PlWoodside, NY 11377",United States,New York,Queens County,Queens,53rd Place,"32-50 53rd Pl, Woodside, NY 11377, USA",40.754361,-73.907795
2175,Brokered by New York Realty Minimax,Multi-family home for sale,2198888,11,6.0,2184.207862,558 Seneca Ave,"Ridgewood, NY 11385","558 Seneca AveRidgewood, NY 11385",United States,New York,Queens County,Queens,Seneca Avenue,"558 Seneca Ave, Ridgewood, NY 11385, USA",40.704719,-73.911227
3465,"Brokered by BMB Living, Inc.",Co-op for sale,410000,3,1.0,766.000000,42 W 138th St Apt 31,"New York, NY 10037","42 W 138th St Apt 31New York, NY 10037",New York,New York County,New York,Manhattan,42,"42 W 138th St Apt 31, New York, NY 10037, USA",40.815359,-73.937473
2756,Brokered by Astor Brokerage Ltd,House for sale,849999,3,2.0,2184.207862,147-20 76th Rd,"Flushing, NY 11367","147-20 76th RdFlushing, NY 11367",United States,New York,Queens County,Queens,76th Road,"147-20 76th Rd, Flushing, NY 11367, USA",40.722634,-73.816566
1904,Brokered by Motivated Access Realty Corp.,Multi-family home for sale,949000,7,3.0,2745.000000,431 Saint Lawrence Ave,"Bronx, NY 10473","431 Saint Lawrence AveBronx, NY 10473",United States,New York,Bronx County,The Bronx,Saint Lawrence Avenue,"431 St Lawrence Ave, Bronx, NY 10473, USA",40.813620,-73.863328


In [6]:
# Список столбцов
df.columns.tolist()

['BROKERTITLE',
 'TYPE',
 'PRICE',
 'BEDS',
 'BATH',
 'PROPERTYSQFT',
 'ADDRESS',
 'STATE',
 'MAIN_ADDRESS',
 'ADMINISTRATIVE_AREA_LEVEL_2',
 'LOCALITY',
 'SUBLOCALITY',
 'STREET_NAME',
 'LONG_NAME',
 'FORMATTED_ADDRESS',
 'LATITUDE',
 'LONGITUDE']

In [7]:
# Типы данных и общая информация
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4801 entries, 0 to 4800
Data columns (total 17 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   BROKERTITLE                  4801 non-null   object 
 1   TYPE                         4801 non-null   object 
 2   PRICE                        4801 non-null   int64  
 3   BEDS                         4801 non-null   int64  
 4   BATH                         4801 non-null   float64
 5   PROPERTYSQFT                 4801 non-null   float64
 6   ADDRESS                      4801 non-null   object 
 7   STATE                        4801 non-null   object 
 8   MAIN_ADDRESS                 4801 non-null   object 
 9   ADMINISTRATIVE_AREA_LEVEL_2  4801 non-null   object 
 10  LOCALITY                     4801 non-null   object 
 11  SUBLOCALITY                  4801 non-null   object 
 12  STREET_NAME                  4801 non-null   object 
 13  LONG_NAME         

In [8]:
# Количество пропусков по столбцам
df.isnull().sum()

BROKERTITLE                    0
TYPE                           0
PRICE                          0
BEDS                           0
BATH                           0
PROPERTYSQFT                   0
ADDRESS                        0
STATE                          0
MAIN_ADDRESS                   0
ADMINISTRATIVE_AREA_LEVEL_2    0
LOCALITY                       0
SUBLOCALITY                    0
STREET_NAME                    0
LONG_NAME                      0
FORMATTED_ADDRESS              0
LATITUDE                       0
LONGITUDE                      0
dtype: int64

In [9]:
# Количество полных дубликатов строк
df.duplicated().sum()

214

In [10]:
# Количество уникальных значений по столбцам
df.nunique()

BROKERTITLE                    1036
TYPE                             13
PRICE                          1274
BEDS                             27
BATH                             22
PROPERTYSQFT                   1445
ADDRESS                        4582
STATE                           308
MAIN_ADDRESS                   4583
ADMINISTRATIVE_AREA_LEVEL_2      29
LOCALITY                         11
SUBLOCALITY                      21
STREET_NAME                     174
LONG_NAME                      2731
FORMATTED_ADDRESS              4550
LATITUDE                       4196
LONGITUDE                      4118
dtype: int64

In [11]:
# Базовая статистика по числовым столбцам
df.describe()

,PRICE,BEDS,BATH,PROPERTYSQFT,LATITUDE,LONGITUDE
count,4.801000e+03,4801.000000,4801.000000,4801.000000,4801.000000,4801.000000
mean,2.356940e+06,3.356801,2.373861,2184.207862,40.714227,-73.941601
std,3.135525e+07,2.602315,1.946962,2377.140894,0.087676,0.101082
min,2.494000e+03,1.000000,0.000000,230.000000,40.499546,-74.253033
25%,4.990000e+05,2.000000,1.000000,1200.000000,40.639375,-73.987143
50%,8.250000e+05,3.000000,2.000000,2184.207862,40.726749,-73.949189
75%,1.495000e+06,4.000000,3.000000,2184.207862,40.771923,-73.870638
max,2.147484e+09,50.000000,50.000000,65535.000000,40.912729,-73.702450


## 20 простых вопросов для исследования данных

In [14]:
#  Самая высокая цена
df["PRICE"].max()

2147483647

In [15]:
#  Самая низкая цена
df["PRICE"].min()

2494

In [16]:
#  Топ-10 районов по числу объявлений
df["SUBLOCALITY"].value_counts().head(10)

SUBLOCALITY
New York           1016
Kings County        727
Queens County       684
Queens              555
Richmond County     480
Brooklyn            464
Bronx County        307
New York County     286
The Bronx           184
Staten Island        59
Name: count, dtype: int64

In [17]:
#  Средняя цена по районам
df.groupby("SUBLOCALITY")["PRICE"].mean()

SUBLOCALITY
Bronx County        1.020866e+06
Brooklyn            8.646435e+05
Brooklyn Heights    6.250000e+05
Coney Island        5.113333e+05
Dumbo               5.799000e+06
East Bronx          2.650000e+05
Flushing            4.760000e+05
Fort Hamilton       5.990000e+05
Jackson Heights     9.850000e+05
Kings County        1.795465e+06
Manhattan           3.609905e+06
New York            2.534033e+06
New York County     1.019120e+07
Queens              4.437331e+05
Queens County       1.135774e+06
Rego Park           2.150000e+05
Richmond County     5.427598e+06
Riverdale           3.470000e+05
Snyder Avenue       6.500000e+05
Staten Island       4.475819e+05
The Bronx           3.366162e+05
Name: PRICE, dtype: float64

In [18]:
#  Сколько домов дороже 1 млн (фильтр)
(df["PRICE"] > 1_000_000).sum()

1839

In [19]:
#  Средняя площадь (кв. футы)
df["PROPERTYSQFT"].mean()

2184.207861758384

In [20]:
#  Самый большой объект по площади
df.loc[df["PROPERTYSQFT"].idxmax(), ["PROPERTYSQFT", "PRICE", "ADDRESS"]]

PROPERTYSQFT        65535.0
PRICE              31500000
ADDRESS         467 Keap St
Name: 4623, dtype: object

In [21]:
#  Средняя цена за кв. фут
(df["PRICE"] / df["PROPERTYSQFT"]).mean()

823.0372562356092

In [22]:
# Средняя цена по числу спален
df.groupby("BEDS")["PRICE"].mean()

BEDS
1     5.337822e+05
2     1.046006e+06
3     1.462133e+06
4     2.653452e+06
5     4.150148e+06
6     3.337154e+06
7     2.717450e+07
8     5.443654e+06
9     2.080872e+06
10    5.539555e+06
11    5.129839e+06
12    2.998478e+06
13    2.027800e+06
14    1.650000e+06
15    2.137777e+06
16    1.993500e+06
18    2.499000e+06
19    1.998000e+06
20    5.698000e+06
24    3.066667e+06
30    3.500000e+06
32    4.000000e+06
35    4.380000e+06
36    3.099000e+06
40    5.980000e+06
42    1.070000e+07
50    1.100000e+07
Name: PRICE, dtype: float64

In [23]:
#  Связь цены, спален, ванных и площади
df[["PRICE", "BEDS", "BATH", "PROPERTYSQFT"]].corr()

,PRICE,BEDS,BATH,PROPERTYSQFT
PRICE,1.000000,0.052189,0.079371,0.110889
BEDS,0.052189,1.000000,0.776447,0.420503
BATH,0.079371,0.776447,1.000000,0.483935
PROPERTYSQFT,0.110889,0.420503,0.483935,1.000000


In [24]:
# Сколько уникальных городов (LOCALITY)
df["LOCALITY"].nunique()

11

In [25]:
#  Дома с 3 спальнями дешевле 500 тыс 
df[(df["BEDS"] == 3) & (df["PRICE"] < 500_000)]

,BROKERTITLE,TYPE,PRICE,BEDS,BATH,PROPERTYSQFT,ADDRESS,STATE,MAIN_ADDRESS,ADMINISTRATIVE_AREA_LEVEL_2,LOCALITY,SUBLOCALITY,STREET_NAME,LONG_NAME,FORMATTED_ADDRESS,LATITUDE,LONGITUDE
3,Brokered by COMPASS,Condo for sale,69000,3,1.000000,445.000000,2 E 55th St Unit 908W33,"Manhattan, NY 10022","2 E 55th St Unit 908W33Manhattan, NY 10022",United States,New York,New York County,New York,East 55th Street,"2 E 55th St, New York, NY 10022, USA",40.761398,-73.974613
12,Brokered by COMPASS,Co-op for sale,259000,3,1.000000,2184.207862,34-41 85th St Unit 1D,"Jackson Heights, NY 11372","34-41 85th St Unit 1DJackson Heights, NY 11372",New York,Queens County,Queens,Flushing,34-41,"34-41 85th St #1b, Flushing, NY 11372, USA",40.753119,-73.881876
17,Brokered by Dorsa Group Realty,Co-op for sale,199000,3,1.000000,325.000000,9430 Ridge Blvd Apt 6D,"Brooklyn, NY 11209","9430 Ridge Blvd Apt 6DBrooklyn, NY 11209",New York,Kings County,Brooklyn,Fort Hamilton,9430,"9430 Ridge Blvd #6d, Brooklyn, NY 11209, USA",40.618103,-74.036905
78,Brokered by Meyers Enterprise Realty Group,Land for sale,119000,3,2.373861,2184.207862,3652 DE Reimer Ave,"Bronx, NY 10466","3652 DE Reimer AveBronx, NY 10466",United States,New York,Bronx County,The Bronx,De Reimer Avenue,"3652 De Reimer Ave, Bronx, NY 10466, USA",40.884170,-73.838839
91,Brokered by Sowae Corp,House for sale,314000,3,1.000000,1572.000000,850 E 39th St,"Brooklyn, NY 11210","850 E 39th StBrooklyn, NY 11210",United States,New York,Kings County,Brooklyn,East 39th Street,"850 E 39th St, Brooklyn, NY 11210, USA",40.636237,-73.939679
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
4727,Brokered by Homefinders of SI Inc.,Land for sale,225000,3,2.373861,2184.207862,229 Mosely Ave,"Staten Island, NY 10312","229 Mosely AveStaten Island, NY 10312",United States,New York,Richmond County,Staten Island,Mosely Avenue,"229 Mosely Ave, Staten Island, NY 10312, USA",40.541076,-74.170620
4730,Brokered by Douglas Elliman - 280 Metropolitan,Co-op for sale,378023,3,1.000000,2184.207862,301 E 22nd St Apt 2G,"New York, NY 10010","301 E 22nd St Apt 2GNew York, NY 10010",New York,New York County,New York,Manhattan,301,"301 E 22nd St #2g, New York, NY 10010, USA",40.737125,-73.980661
4764,NoBroker,Co-op for sale,189000,3,1.000000,300.000000,460 Ovington Ave Apt 2H,"Brooklyn, NY 11209","460 Ovington Ave Apt 2HBrooklyn, NY 11209",New York,Kings County,Brooklyn,Bay Ridge,460,"460 Ovington Ave #2h, Brooklyn, NY 11209, USA",40.632771,-74.022251
4787,Brokered by REAL NEW YORK - Remote Brooklyn,Condo for sale,499000,3,1.000000,472.000000,127 W 112th St Apt 1B,"New York, NY 10026","127 W 112th St Apt 1BNew York, NY 10026",New York,New York County,New York,Manhattan,127,"127 W 112th St #1b, New York, NY 10026, USA",40.800382,-73.953136


In [26]:
# Средняя цена и площадь по типу объекта
df.groupby("TYPE")[["PRICE", "PROPERTYSQFT"]].mean()

,PRICE,PROPERTYSQFT
TYPE,,
Co-op for sale,1.100418e+06,1661.939812
Coming Soon,1.172000e+06,2217.103931
Condo for sale,2.630710e+06,1574.625033
Condop for sale,9.986000e+05,2184.207862
Contingent,8.825717e+05,1775.223442
For sale,1.954536e+06,2116.483145
Foreclosure,1.343010e+06,2331.229133
House for sale,3.684216e+06,2255.556856
Land for sale,1.073021e+06,2159.019946


In [27]:
#  В каком районе жильё в среднем самое дорогое?
df.groupby("SUBLOCALITY")["PRICE"].median().sort_values(ascending=False).head(10)

SUBLOCALITY
New York County    6497500.0
Dumbo              5799000.0
Manhattan          2795000.0
Kings County       1249000.0
New York           1025000.0
Jackson Heights     985000.0
Queens County       948500.0
Bronx County        795000.0
Richmond County     750000.0
Snyder Avenue       650000.0
Name: PRICE, dtype: float64

In [28]:
#  Какой тип жилья самый дорогой, а какой доступный?
df.groupby("TYPE")["PRICE"].median().sort_values(ascending=False)

TYPE
Townhouse for sale            2950000.0
Mobile house for sale         1288000.0
Multi-family home for sale    1199000.0
Coming Soon                   1172000.0
Condop for sale               1080000.0
For sale                      1044500.0
Condo for sale                 899000.0
House for sale                 859000.0
Pending                        800000.0
Contingent                     682450.0
Land for sale                  650000.0
Foreclosure                    592450.0
Co-op for sale                 425000.0
Name: PRICE, dtype: float64

In [29]:
#  Как растёт цена с числом спален?
df.groupby("BEDS")["PRICE"].median().head(8)

BEDS
1     395000.0
2     670000.0
3     750000.0
4    1149500.0
5    1399000.0
6    1399000.0
7    1690000.0
8    1874500.0
Name: PRICE, dtype: float64

In [30]:
#  В каких районах больше всего предложений на рынке?
df["SUBLOCALITY"].value_counts().head(10)

SUBLOCALITY
New York           1016
Kings County        727
Queens County       684
Queens              555
Richmond County     480
Brooklyn            464
Bronx County        307
New York County     286
The Bronx           184
Staten Island        59
Name: count, dtype: int64

In [31]:
# Средняя цена за квадратный фут по районам
df["price_per_sqft"] = df["PRICE"] / df["PROPERTYSQFT"]
df.groupby("SUBLOCALITY")["price_per_sqft"].mean().sort_values(ascending=False).head(10)

SUBLOCALITY
Dumbo               2322.386864
New York County     2310.951797
Manhattan           1821.147129
New York            1198.615854
Brooklyn Heights     958.588957
Richmond County      903.284357
Fort Hamilton        767.948718
Kings County         709.825461
Brooklyn             660.699305
Coney Island         623.147457
Name: price_per_sqft, dtype: float64

## Краткое саммари

**Объём данных:** 4801 строка, 17 столбцов. Пропусков нет, но есть **214 полных дубликатов** (стоит удалить через `df.drop_duplicates()`).

**Основные выводы:**
- **Цены** сильно разбросаны: медиана ≈ 825 тыс., среднее ≈ 2.36 млн — распределение скошено вправо из-за дорогих объектов.
- **Самые дорогие районы** (по медиане): New York County (~6.5 млн), Dumbo (~5.8 млн), Manhattan (~2.8 млн). Самые доступные — Queens, The Bronx, Staten Island.
- **По типу жилья**: дороже всего Townhouse (медиана ~2.95 млн), доступнее всего Co-op (~425 тыс.).
- **Цена растёт с числом спален** (медиана от 395 тыс. за 1 спальню до 1.87 млн за 8), но связь нелинейная.
- **Цена за кв. фут** максимальна в Dumbo, New York County и Manhattan (~1800–2300 $/sqft).
- **Корреляции** PRICE с BEDS/BATH/PROPERTYSQFT слабые (0.05–0.11) — в основном из-за выбросов.
- **Больше всего предложений** в New York, Kings County, Queens County.

**⚠️ Замечания по качеству данных (видны из запросов):**
- `PRICE` max = **2 147 483 647** (это максимум int32) — явно ошибочное/заглушечное значение, искажает среднее.
- `PROPERTYSQFT` = **2184.207862** многократно повторяется — похоже на значение-заполнитель (импутация средним).
- `BATH` содержит дробные значения вроде **2.373861** (то же среднее) — тоже признак заполнения пропусков.
- Категории районов дублируются по смыслу (`New York`/`New York County`/`Manhattan`, `Kings County`/`Brooklyn`) — стоит унифицировать перед группировкой.

**Рекомендации для дальнейшего анализа:** удалить дубликаты, отфильтровать аномальные цены и заглушки в площади/ваннах, привести категории районов к единому виду.
